### ISSUE TITLE
command line prime function -in filename

### ISSUE THREAD

**Issue by maroonbells** [openssl/openssl#20357]
feature suggestion for the command line 'prime' function in addition to #19208

Unless the ability is already there and I'm not seeing it - if there can be a way for 'prime' to obtain the prime number from a disk file, for numbers too long for the command line.

**Comment by paulidale** [comment]
For Unix like shells: `openssl prime $(cat -n file)`

**Comment by yogi-ji** [comment]
Hello, can I work on this as my first contribution ?
If I understood it correctly, we have to add -in parameter to "openssl prime" operation which accepts file, right ?

**Comment by paulidale** [comment]
That's the suggestion.

**Comment by maroonbells** [comment]
Yes, with the intent that it's a way to support numbers larger than can fit in the Windows command line.
If it helps to have a sample to work with, you could use one of the Mersenne primes, such as 2^9689-1, which is the hex string of '1' followed by 2422 "F" digits.

**Comment by yogi-ji** [comment]
@maroonbells thank you for clarifying that, I will start working on it

**Comment by t8m** [comment]
Please note that the BIGNUM implementation in OpenSSL is not supposed to be able to handle arbitrarily large numbers - especially some computations that handle private keys or key parameters expect the input to be "reasonably sized" in terms of values being used in usual algorithms such as RSA DSA DH or EC. Especially the consttime operation is not guaranteed to work or the operations might completely fail if the values are bigger than that.

**Comment by zriback** [comment]
Looks like this never got resolved? Can I work on a solution for this?

**Comment by ritesh006** [comment]
is this issue still relevant?

**Comment by anatoliiohorodnyk** [comment]
@ritesh006 https://github.com/openssl/openssl/pull/26549 [Merged](https://github.com/openssl/openssl/pull/26549#issuecomment-2837411746)
